src/lib/stack-internal.js

```diff
@@ -0,0 +1,201 @@
+/* global __TARGET__ */
+/* eslint-env browser */
+
+import getIcon from './icon'
+import initializeRealtime from './realtime'
+import normalizeURL from './normalize-url'
+
+import {
+  ForbiddenException,
+  ServerErrorException,
+  NotFoundException,
+  MethodNotAllowedException,
+  UnavailableStackException,
+  UnavailableSettingsException,
+  UnauthorizedStackException
+} from './exceptions'
+
+// the option credentials:include tells fetch to include the cookies in the
+// request even for cross-origin requests
+function fetchOptions() {
+  return {
+    credentials: 'include',
+    headers: {
+      Authorization: `Bearer ${COZY_TOKEN}`,
+      Accept: 'application/json'
+    }
+  }
+}
+
+let COZY_URL
+let COZY_HOST
+let COZY_TOKEN
+let USE_SSL
+
+const errorStatuses = {
+  '401': UnauthorizedStackException,
+  '403': ForbiddenException,
+  '404': NotFoundException,
+  '405': MethodNotAllowedException,
+  '500': ServerErrorException
+}
+
+function getApps() {
+  return fetchJSON(`${COZY_URL}/apps/`, fetchOptions()).then(json => {
+    if (json.error) throw new Error(json.error)
+    else return json.data
+  })
+}
+
+function fetchJSON(url, options) {
+  return fetch(url, options).then(res => {
+    if (typeof errorStatuses[res.status] === 'function') {
+      throw new errorStatuses[res.status]()
+    }
+
+    return res.json()
+  })
+}
+
+// fetch function with the same interface than in cozy-client-js
+function cozyFetchJSON(cozy, method, path, body) {
+  const requestOptions = Object.assign({}, fetchOptions(), {
+    method
+  })
+  requestOptions.headers['Accept'] = 'application/json'
+  if (method !== 'GET' && method !== 'HEAD' && body !== undefined) {
+    if (requestOptions.headers['Content-Type']) {
+      requestOptions.body = body
+    } else {
+      requestOptions.headers['Content-Type'] = 'application/json'
+      requestOptions.body = JSON.stringify(body)
+    }
+  }
+  return fetchJSON(`${COZY_URL}${path}`, requestOptions).then(json => {
+    const responseData = Object.assign({}, json.data)
+    if (responseData.id) responseData._id = responseData.id
+    return Promise.resolve(responseData)
+  })
+}
+
+function getStorageData() {
+  return fetchJSON(`${COZY_URL}/settings/disk-usage`, fetchOptions())
```

Comment:
I'm curious: does it work? It sending `Accept: application/json` instead of the [documented](https://docs.cozy.io/en/cozy-stack/settings/#disk-usage) `Accept: application/vnd.api+json` but the stack often accepts both.

Reply:
It does, and as this is legacy, I wouldn't touch it in this PR (but we could do that in another one if you thinks its usefull)